Project to check

romankostyuk opened the pull request.

onely committed:
Project to check

nikolay-977 commented on src/main/java/model/MainPage.java:
```diff
@@ -0,0 +1,61 @@
+import java.time.Duration;
+public class MainPage {
+    private final WebDriver driver;
+    private static final By CHECK_STATUS_BUTTON = By.xpath(".//button[@class=\"Header_Link__1TAG7\"]");
```
Необходимо исправить: В теме UI-тестирование > Page Object есть примеры использования локаторов и методов(шагов). В данных примерах не используется ключевое слово static. Не следует его использовать для локаторов и шагов при разработке автотестов, т.к. java это объектно ориентированный язык и важно научиться работать с объектами, а не только с классами.

nikolay-977 commented on src/main/java/model/MainPage.java:
```diff
@@ -0,0 +1,61 @@
+    private static final By CREATE_ORDER_BUTTON_TOP = By.xpath(".//button[@class='Button_Button__ra12g']");
+    private static final By CREATE_ORDER_BUTTON_BOTTOM = By.cssSelector("div > button[class=\"Button_Button__ra12g Button_UltraBig__UU3Lp\"]");
+
+    public MainPage(WebDriver driver){
```
Можно улучшить: для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea используй сочетание клавиш Ctrl+Alt+L. Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 commented on src/test/java/FaqTest.java:
```diff
@@ -0,0 +1,60 @@
+        this.numberOfQuestion = numberOfQuestion;
+        this.expectedAnswerText = expectedAnswerText;
+    }
+    @Parameterized.Parameters
```
Можно улучшить: В параметризованных тестах для аннотации `@Parameterized.Parameters` лучше использовать аргумент name: `@Parameterized.Parameters(name = "Стоимость булочки. Тестовые данные: {0} {1}")`, где {0}, {1} - индексы параметров. С аргументом name наименование теста отобразится в отчете. А использование индексов параметров повысит информативность проверки.

nikolay-977 commented on pom.xml:
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (mvn clean test). Как оформлять README.md можно ознакомиться в статье https://dzen.ru/a/YbzudflBPE1IDJF-

nikolay-977 commented on src/test/java/OrderTest.java:
```diff
@@ -0,0 +1,83 @@
+        MainPage mainPage = new MainPage(driver);
+        mainPage.open();
+        OrderPage orderPage = mainPage.clickCreateOrderButtonTop();
+        orderPage.fillNames(lastName, firstName);
```
Можно улучшить: В теме UI-тестирование > Page Object > Работа с методами > Объединение методов хорошо описан прием объединения методов. Применяй его для сокращения кода автотеста, где это необходимо. Например, при заполнении форм.

nikolay-977 commented on src/test/java/OrderTest.java:
```diff
@@ -0,0 +1,83 @@
+    public void sendOrder_validData_confirmedMessage() {
+        MainPage mainPage = new MainPage(driver);
+        mainPage.open();
+        OrderPage orderPage = mainPage.clickCreateOrderButtonTop();
```
Необходимо исправить: Обрати внимание, что есть две точки входа в сценарий: кнопка «Заказать» вверху страницы и внизу. Нужно реализовать оба кейса.

nikolay-977 commented on src/test/java/OrderTest.java:
```diff
@@ -0,0 +1,83 @@
+        orderPage.fillComment(comment);
+        orderPage.clickCreateOrderButton();
+        orderPage.clickConfirmButton();
+        Assert.assertTrue(orderPage.isOrderConfirmed());
```
Необходимо исправить: Проверки в тестовых методах должны быть в одном стиле. Когда код написан в одном стиле, это улучшает читаемость кода, сокращает время локализации дефекта и упрощает его поддержку. Не следует в тестовых методах одновременно использовать проверки из разных фреймворков org.hamcrest.MatcherAssert(метод assertThat) и org.junit.Assert(методы assertEquals, assertTrue и т.д.). В этих фреймворках методы похожие, но имеют разные сигнатуры. Есть вероятность допустить ошибку.  В этих фреймворках отличается формат сообщений об ошибках в случае фейла, что не очень удобно при локализации дефектов.

onely committed:
1. Исправил расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea
2. Убрал static для локаторов
3. Добавил в параметризованные тесты аргумент name с индексами параметров для большей информативности.
4. Объеденил заполнение форм в один метод в тестах заказа.
5. Параметризовал локатор кнопок заказать для теста заказа, теперь в первом тесте верхняя кнопка, во втором - нижняя.
6. Теперь использую проверки из одного фреймворка org.junit.Assert
7. Добавил простенький readme
